Automattic/jetpack
# AI Assistant: compute feature availability optimistically

When the site archives the requests limit, depending on the current plan/tier,m we should compute if the limit has been achieved, and if the site requires an upgrade. This PR takes over the `isOverLimit` and `requiredUpgrade` props, based on the server-side computing. 

For now, it works with the default plan.

Fixes #

## Proposed changes:

* AI Assistant: compute feature availability optimistically

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Test with a site with a Free plan
* Go to the block editor
* Open the Jetpack AI side
* Identify the Usage panel
* Open the Redux dev tool
* Identify the `wordpress-com/plans` store
* Open the dev console
* Let's store the testing state. Copy the following code and paste it into your dev console:

```js
wp.data.dispatch( 'wordpress-com/plans' ).storeAiAssistantFeature( {
    "hasFeature": false,
    "isOverLimit": false,
    "requestsCount": 15,
    "requestsLimit": 20,
    "requireUpgrade": false,
    "upgradeType": "default",
    "usagePeriod": {
        "currentStart": "2023-11-06 00:00:00",
        "nextStart": "2023-12-06 00:00:00",
        "requestsCount": 15
    },
    "currentTier": {
        "value": 1
    },
    "nextTier": {
        "slug": "ai-assistant-tier-100",
        "limit": 100,
        "value": 100
    }
} )
```
* Confirm the Usage panel reflects the right stored
* Increase the requests counter by dispatching the action:

```js
wp.data.dispatch( 'wordpress-com/plans' ).increaseAiAssistantRequestsCount()
```
* Confirm the request counter increases
* Confirm the panel changes the bar color and info once the limit is achieved


https://github.com/Automattic/jetpack/assets/77539/7d87d230-3ce2-41f5-af8c-311476b03a6e
